### PR TITLE
 Add a fallback test for invalid server name

### PR DESCRIPTION
**Pull request by AspirinSJL**
If the balancer can't find any backend for the server name in the load balance request, the balancer will return `NOT_FOUND` status to the client. Re-resolution and re-connection may happen, but won't help. Eventually, the client will enter fallback mode. 

This test is added upon request from @a11r .

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] Performance differences noted:
Benchmark                    cpu_time    real_time
---------------------------  ----------  -----------
BM_MetadataRefUnrefExternal  -5%         -5%
BM_MetadataRefUnrefStatic    -5%         -5%
```

File: include/grpcpp/impl/codegen/status.h [hunk 1]
```diff
@@ -105,6 +105,8 @@ class Status {
   static const Status& OK;
   /// A CANCELLED pre-defined instance.
   static const Status& CANCELLED;
+  /// A NOT_FOUND pre-defined instance.
+  static const Status& NOT_FOUND;
```
Comment (vjpai) [inline include/grpcpp/impl/codegen/status.h]: gRFC?

Comment (AspirinSJL) [reply include/grpcpp/impl/codegen/status.h]: Reverted.

**Comment by AspirinSJL** [comment]
@vjpai Thanks for pointing this out. I have removed that addition from include.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by markdroth** [comment]
My understanding from our internal discussion is that we decided that the balancer should not return `NOT_FOUND` in this case.  Instead, it will send a serverlist containing a single drop entry and then leave the stream idle.  Given that, this test case is not actually testing what will occur when the server name is unknown.

Furthermore, since the code does not treat `NOT_FOUND` any differently than any other status, I believe the functionality covered by this test is already covered by existing fallback tests.

Given those facts, I don't think this test case is actually needed.

**Comment by AspirinSJL** [comment]
No longer needed.

**Pull request closed without merging**